### ISSUE TITLE
bug while turning off automation condition sensor through UI

### ISSUE THREAD

**Issue by rmalhotra2192** [dagster-io/dagster#23648]
### Dagster version

1.8.0

### What's the issue?

An unexpected GraphQL error occurred when attempting to toggle a sensor in Dagster's UI. The error prevents the sensor's status from updating, and the issue appears sporadically when interacting with the sensor list.

<img width="1432" alt="Screenshot 2024-08-14 at 9 26 37 PM" src="https://github.com/user-attachments/assets/f382ef9e-6317-41aa-8ac2-82842f4925e4">

<img width="1434" alt="Screenshot 2024-08-14 at 9 27 26 PM" src="https://github.com/user-attachments/assets/9e332a0f-099e-48d7-88c1-c24ab645a30e">

### What did you expect to happen?

The existing behavior is incorrect because the UI should allow users to toggle sensors without encountering a GraphQL error.

### How to reproduce?

- Open the Dagster dashboard.
- Navigate to the Sensors tab.
- Attempt to toggle the default_automation_condition_sensor on or off.

### Deployment type

Docker Compose

### Deployment details

This is a complete Open Source deployment. Deployed on AWS ECS with docker-compose.

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by garethbrickman** [comment]
Duplicates https://github.com/dagster-io/dagster/issues/23567

A fix will be included in the upcoming 1.8.1 release this week.

**Comment by liongkj** [comment]
```bash
Operation name: {SingleSensorQuery}

Message: (psycopg2.errors.QueryCanceled) canceling statement due to statement timeout

[SQL: SELECT runs.id, runs.run_body, runs.status, runs.create_timestamp, runs.update_timestamp, runs.start_time, runs.end_time 
FROM runs JOIN run_tags AS run_tags_filter0 ON runs.run_id = run_tags_filter0.run_id AND run_tags_filter0.key = %(key_1)s AND run_tags_filter0.value = %(value_1)s JOIN run_tags AS run_tags_filter1 ON runs.run_id = run_tags_filter1.run_id AND run_tags_filter1.key = %(key_2)s AND run_tags_filter1.value = %(value_2)s ORDER BY runs.id DESC 
 LIMIT %(param_1)s]
[parameters: {'key_1': 'dagster/sensor_name', 'value_1': 'default_automation_condition_sensor', 'key_2': '.dagster/repository', 'value_2': '__repository__@thingsboard_code', 'param_1': 1}]
(Background on this error at: https://sqlalche.me/e/20/e3q8)

Path: ["sensorOrError","sensorState","runs"]

Locations: [{"line":30,"column":9}]
```


i am still seeing this error after upgrading to 1.8.1. Can you advise what to do?

**Comment by rmalhotra2192** [comment]
original issue fixed in 1.8.1. this seems like a seperate issue.

> ```shell
> Operation name: {SingleSensorQuery}
> 
> Message: (psycopg2.errors.QueryCanceled) canceling statement due to statement timeout
> 
> [SQL: SELECT runs.id, runs.run_body, runs.status, runs.create_timestamp, runs.update_timestamp, runs.start_time, runs.end_time 
> FROM runs JOIN run_tags AS run_tags_filter0 ON runs.run_id = run_tags_filter0.run_id AND run_tags_filter0.key = %(key_1)s AND run_tags_filter0.value = %(value_1)s JOIN run_tags AS run_tags_filter1 ON runs.run_id = run_tags_filter1.run_id AND run_tags_filter1.key = %(key_2)s AND run_tags_filter1.value = %(value_2)s ORDER BY runs.id DESC 
>  LIMIT %(param_1)s]
> [parameters: {'key_1': 'dagster/sensor_name', 'value_1': 'default_automation_condition_sensor', 'key_2': '.dagster/repository', 'value_2': '__repository__@thingsboard_code', 'param_1': 1}]
> (Background on this error at: https://sqlalche.me/e/20/e3q8)
> 
> Path: ["sensorOrError","sensorState","runs"]
> 
> Locations: [{"line":30,"column":9}]
> ```
> 
> i am still seeing this error after upgrading to 1.8.1. Can you advise what to do?
